development/components/faceted-search/inside-faceted-search-module.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Inside the module
+weight: 30
+---
+
+# Inside the module
+
+This page should be useful for someone willing to contribute to the module. To do so he/she needs to understand the module logic and structure.
+
+## Exposed endpoints
+
+The module
+
+- is accessible from the front-office on dedicated pages, using hooks
+- has a back-office configuration page
+- provides 3 endpoints to be regularly visited (by a cron for example) in order to reindex the products
+
+The 3 exposed endpoints are in the root folder of the module.
+
+Indexation results are stored in specific SQL tables. This allows to query flat tables ready for querying, instead of having complex SQL queries being fired at runtime when user is searching on front-office.
+
+## Filter system
+
+On front-office, user can see multiple blocks. Blocks are linked to a filter and the population. The population is the products matching the filter.
+
+Blocks are what is the most expensive to generate. Each link is a query.
+
+Filters have a complex structure allowing deep queries such as OR and AND, under, equals, above... and at the end of the processing SQL queries are built from them to be run against the database.
+
+QUESTION: where are the Filters objects? they are arrays ?
```

Comment:
Filters are objects: https://github.com/PrestaShop/ps_facetedsearch/blob/dev/src/Filters/Converter.php#L188

Facets too: https://github.com/PrestaShop/ps_facetedsearch/blob/dev/src/Filters/Converter.php#L103

A facet is for example Color, filter is Red, Blue, White, Black, etc.